docs/spec/v1beta2/helmcharts.md

```diff
@@ -390,6 +390,54 @@ Besides being reported in Events, the reconciliation errors are also logged by
 the controller. The Flux CLI offer commands for filtering the logs for a
 specific HelmChart, e.g. `flux logs --level=error --kind=HelmChart --name=<chart-name>`.
 
+### Improving Resource Consumption Using The Cache
+
+When using a Helm Repository as source for a Helm chart and dependencies, the
+controller loads the Helm Repository index in-memory to find the latest version
+of the chart.
+
+The controller can be configured to cache a Helm Repository indexes in-memory.
+The cache is used to avoid loading Helm Repository indexes for every Helm chart
+reconciliation.
+
+Three fields are provided to enable and configure the cache:
+- `helm-cache-max-size`: The maximum size of the cache in number of indexes.
+  If `0`, then the cache is disabled.
+- `helm-cache-ttl`: The TTL of an index in the cache.
+- `helm-cache-purge-interval`: The interval at which the cache is purged of
+  expired items. 
+
+The caching strategy is to pull a Helm Repository index from the cache if it is 
+available, otherwise to load the index, to retrieve and build the chart,
```

Comment:
```suggestion
available, otherwise to load the index, retrieve and build the chart,
```

Implementation detail: I am wondering if it would be better to cache it as soon as we have successfully parsed the YAML, this to make it available to other processes as quickly as possible.

Reply:
yes that would be nice. Keeping this for the follow up.